enterprise/backend/src/metabase_enterprise/advanced_permissions/driver/impersonation.clj

```diff
@@ -70,10 +70,17 @@
                 role-attribute     (:attribute conn-impersonation)
                 user-attributes    (:login_attributes @api/*current-user*)
                 role               (get user-attributes role-attribute)]
-            (if (str/blank? role)
+            (cond
+              (and (some? role) (not (string? role)))
```

Comment:
nit: `nil` is not a string so I believe the `some?` is redundant here

Reply:
still nitting, dont take it as a blocker: I see why you want it (so we skip over the 2nd branch in the cond). I think it might be clearer to make `(not (string? role))` the 2nd cond branch predicate, and start with checking `(str/blank? role)`.